data/network/src/commonMain/kotlin/social/androiddev/common/network/MastodonApiImpl.kt

```diff
@@ -58,4 +44,20 @@ class MastodonApiImpl(
                 }.body()
         }
     }
+
+    override suspend fun verifyApplication(): Result<Application> {
+        return runCatchingIgnoreCancelled<Application> {
+            httpClient.get("/api/v1/apps/verify_credentials") {
+                headers {
+                    append(HttpHeaders.Authorization, "testAuthorization")
+                }
+            }.body()
+        }
+    }
+
+    override suspend fun getInstance(domain: String?): Result<Instance> {
```

Comment:
`domain` shouldn't be null here I believe. but we can fix this later when we hook this up